packages/next-app/src/app/metadata/(components)/SearchPoolForm.tsx

```diff
@@ -0,0 +1,121 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+import React from "react";
+import { Controller, useForm } from "react-hook-form";
+import { useAccount, useNetwork } from "wagmi";
+
+import { Button } from "#/components/Button";
+import { Input } from "#/components/Input";
+import { Select, SelectItem } from "#/components/Select";
+import balancerGql, { networkIdEnumMap } from "#/lib/gql";
+
+export interface PoolMetadataAttribute {
+  poolId: string;
+  network: string;
+}
+
+const inputTypenames = [
+  { value: "1", label: "Mainnet" },
+  { value: "137", label: "Polygon" },
+  { value: "42161", label: "Arbitrum" },
+  { value: "5", label: "Goerli" },
+];
+
+export default function SearchPoolForm({ close }: { close?: () => void }) {
+  const {
+    register,
+    handleSubmit,
+    setError,
+    control,
+    watch,
+    clearErrors,
+    resetField,
+    formState: { errors },
+  } = useForm<PoolMetadataAttribute>();
+  const { push } = useRouter();
+  const { chain } = useNetwork();
+  const { isConnected } = useAccount();
+
+  const poolId = watch("poolId");
+  const network = isConnected ? chain?.id.toString() : watch("network");
+
+  const { data: poolAddress } = balancerGql(network || "1").usePoolAddress(
+    {
+      poolId,
+    },
+    { revalidateIfStale: true }
+  );
+
+  const isPool = poolAddress?.pool?.address ? true : false;
+
+  function handleSubmitForm(formData: PoolMetadataAttribute) {
+    const networkId = formData.network ?? chain?.id.toString();
+    const networkName =
+      networkIdEnumMap[networkId as keyof typeof networkIdEnumMap];
+    push(`/metadata/${networkName}/pool/${formData.poolId}`);
+
+    close?.();
+  }
+
+  React.useEffect(() => {
+    if (poolAddress && !poolAddress.pool && poolId) {
+      setError(
+        "poolId",
+        {
+          type: "notfound",
+          message: "Pool not found. Check the Pool ID and network.",
+        },
+        { shouldFocus: true }
+      );
+      return;
+    } else {
+      clearErrors("poolId");
+    }
+  }, [poolAddress]);
+
+  React.useLayoutEffect(() => {
+    resetField("poolId");
+  }, [network]);
+
+  return (
+    <form onSubmit={handleSubmit(handleSubmitForm)} className="px-2 pt-2">
+      {!isConnected && (
```

Comment:
We'll be effectively making the commit make a bunch more clicks to switch network if we do this. Couldn't we always show this select?

Reply:
> lgtm, trust your call but I don't fully grasp why add the further complication of having the user change networks when searching pools with to me is the most natural flow. Did you have any issues trying to do that?

@ribeirojose I faced some errors when the user chose a pool from a chain that was not the same as the selected chain. However, I am currently devising a solution to address this issue